src/rust/demikernel/libos/network.rs

```diff
@@ -66,7 +66,7 @@ impl NetworkLibOS {
             #[cfg(feature = "catpowder-libos")]
             NetworkLibOS::Catpowder(libos) => libos.socket(domain, socket_type, protocol),
             #[cfg(all(feature = "catnap-libos"))]
-            NetworkLibOS::Catnap(libos) => libos.socket(domain, socket_type, protocol),
+            NetworkLibOS::Catnap(libos) => libos.socket(domain.into(), socket_type.into(), protocol.into()),
```

Comment:
Once you move all LibOSes to this, I would sugges us to move this conversion logic up to the PDPIX layer. What do you think?

Reply:
Yes, definitely! I would love to get rid of some of this code at the libos/network.rs level